salesforce/metadataApi.go

```diff
@@ -0,0 +1,184 @@
+package salesforce
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/amp-labs/connectors/common"
+	"golang.org/x/oauth2"
+)
+
+const (
+	versionPrefix = "v"
+)
+
+var (
+	ErrCreateMetadata  = fmt.Errorf("error in CreateMetadata")
+	ErrCreatingRequest = fmt.Errorf("error in creating request")
+)
+
+func (c *Connector) CreateMetadata(ctx context.Context, operation *common.XMLData, tok *oauth2.Token) (string, error) {
```

Comment:
Can we name the second parameter something other than operation? Since Operation means something else in our backend and API, this might get confusing.